### PR TITLE
Updated examples

### PR DESCRIPTION
**Pull request by arwelHughes**
Makes a new set of examples for using the toolbox.

### CODE REVIEW

**Review by DrPaulSharp — COMMENTED** [review]
All looks good in general, but a bit of tidying up and resolving inconsistencies will help show the code off best.

In addition to the specific comments:
- There is no README for Domains/CustomXY
- There is mixed use of camelCase and snake_case for directories and variables
- The file "Normal_Reflectivity/Custom_XY/customDSPC.m" refers to old variables in the project - is this the correct version of the file?
- "testExamples.m" should be replaced with a new version, rather than just deleted.

File: examples/Domains/Standard_Layers/standardLayersDomainsScript.m [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    'resolution',   'Resolution 1',...
+    'scalefactor',  'Scalefactor 1',...
+    'resample',     false,...
+    'BulkIn',       'SLD Air',...        
+    'BulkOut',      'SLD D2O',...        
+    'domainRatio',  'Domain Ratio 1',....
+    'data',         'Simulation');
+%% 
+% Dinally set the model for our contrast. Each physical contrast must have at 
+% least two domain contrasts and no more.....
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Domains/Standard_Layers/standardLayersDomainsScript.m:100]: @arwelHughes "_at least 2 domain contrasts and no more_" means each contrast must have 2 domains, is this correct?
If so I will reword to "_must have two domain contrasts_"

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply examples/Domains/Standard_Layers/standardLayersDomainsScript.m:100]: I think I'd go with "_must have exactly two domain contrasts_", which feels more unambiguous, and I think was the phrasing used elsewhere.

File: examples/Miscellaneous/Alternative_Languages/orsoDSPCCustomLayers.m [hunk 2]
```diff
@@ -142,3 +144,9 @@
 problem.setContrastModel(1,'DSPC Model');
 problem.setContrastModel(2,'DSPC Model');
 problem.setContrastModel(3,'DSPC Model');
+
+
+%end
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Miscellaneous/Alternative_Languages/orsoDSPCCustomLayers.m:149]: Might have bee used as a function see line 1. I will remove both comments

File: examples/Miscellaneous/Convert_RasCAL1_Project/Model_IIb.m [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+vCH3 = 52.7;
+vCH2 = 28.1;
+vCOO = 39.0;
+vGLYC = 68.8;
+vPO4 = 53.7;
+vCHOL = 120.4;
+vWAT = 30.4;
+
+vHead = vCHOL + vPO4 + vGLYC + 2*vCOO;%320.9;
+vTail = 2*(16*vCH2)+ 2*(vCH3);%1005;
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Miscellaneous/Convert_RasCAL1_Project/Model_IIb.m:76]: I will remove comments are not useful

File: examples/Miscellaneous/absorption/volume_thiolsam_plus_DPPC_polarised_abs.m [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+% total_vol_water = vWAT * Bilayer_wph;
+% 
+% vHead = vHead + total_vol_water;
+% sumb_Head = sumb_Head + total_sumb_water;
+
+
+sldHead = sumb_Head/vHead;
+thickHead = vHead/Bilayer_APM;
+Bil_head_hydr = Bil_head_hydr / 100;
+sldHead = (sldHead * (1 - Bil_head_hydr) +  (Bil_head_hydr * bulk_out(contrast)));
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Miscellaneous/absorption/volume_thiolsam_plus_DPPC_polarised_abs.m:131]: I have changed the variable names to camel case. I have left the function name and comments the same

File: examples/Miscellaneous/absorption/volume_thiolsam_plus_DPPC_polarised_abs.m [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+
+switch contrast
+    case{2,4}
+        output = [ALLOY_UP; GOLD; SAMTAILS; SAMHEAD ; CW ; BILAYER] ;
+    otherwise
+        output = [ALLOY_DOWN; GOLD; SAMTAILS; SAMHEAD ; CW ; BILAYER] ;
+end
+
+sub_rough = Substrate_Roughness;
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Miscellaneous/absorption/volume_thiolsam_plus_DPPC_polarised_abs.m:160]: removed line 160 and modified line 21 to return sub_rough

File: examples/Normal_Reflectivity/Custom_Layers/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+
+
+## Custom Layers Example
+
+This shows an example of using a 'custom layers' model to analyse reflectivity
+from a supported bilayer of DSPC. In essence, custom models allow you to parameterise
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Normal_Reflectivity/Custom_Layers/README.md:6]: DSPC seems to be a well known in acronym in the field. Also first result in search so I will leave as is

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply examples/Normal_Reflectivity/Custom_Layers/README.md:6]: I was expecting it to be easier to spell . . .

File: examples/Normal_Reflectivity/README.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+which case the model is defined as [d, rho, sigma] layers, or to use an external function
+to define the model with any parameterisation using Custom Models. These can either
+be as Custom Layers, where the model ultimately outputs [d, rho, sig] layers
+regardless of the parameterisation, or a 'Custom XY' profile, where the model
+outputs a continuous SLD profile for each contrast.
+
+The examples in the sub-directories are as follows:
+
+- Standard Layers:  This contains an analysis of a DSPC floating bilayer, on
+                    a Silane SAM.
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Normal_Reflectivity/README.md:20]: replaced with self-assembled monolayer (SAM)

File: examples/Domains/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+
+
+## 'Domains Target Functions' Examples
+
+This folder contains examples of how to use RAT to analyse what are referred
+to as 'Domains TF' samples, which are those where an incoherent sum is required
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/Domains/README.md:6]: I will leave this as is, while it is not necessary it causes no confusion as long as we are consistent

File: examples/Domains/Custom_Layers/README.md [hunk 9]
```diff
@@ -8,7 +6,9 @@ layers models.
 These are broadly the same as normal custom layers models, except that the
 input to the custom function contains an additional 'domains' parameter..
```
Comment (DrPaulSharp) [inline examples/Domains/Custom_Layers/README.md:7]: Remove extra full stop here.

File: examples/Domains/Custom_XY/README.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+## Custom XY example Using Domains
+
+This is an example of using incoherent summing ('domains') from custom 
+XY models.
+
+These are broadly the same as normal custom XY models, except that the
+input to the custom function contains an additional 'domains' parameter..
```
Comment (DrPaulSharp) [inline examples/Domains/Custom_XY/README.md:7]: Remove extra full stop here.

File: examples/Domains/README.md [hunk 11]
```diff
@@ -1,20 +1,14 @@
...
+it is possible to use either 'Standard Layers', 'Custom Layers' or 'Custom XY' models.
 
 The basic principle for all three is that there is an additional fittable 
-parameter group called 'domain ratio' (as compared to normal models). 
+parameter group called 'domain ratio' (as compared to normal reflectivity). 
 Then each actual measured contrast, along with the other usual parameters, 
-has a 'Domain Ratio' field. 
+has a 'domain ratio' field. 
 
 The method for defining domains varies between Standard Layers and Custom models,
```
Comment (DrPaulSharp) [inline examples/Domains/README.md:13]: Hmm, maybe write either: standard layers (lower case), or include the inverted commas - 'Standard Layers' - to be consistent with elsewhere.

File: examples/Normal_Reflectivity/Custom_XY/orsoCustomXYDSPC.m [hunk 12]
```diff
@@ -1,19 +1,12 @@
 %% Custom Layers Example for Supported DSPC layer.
-% 
-% 
 % Example of using Custom layers to model a DSPC supportd bilayer.
```
Comment (DrPaulSharp) [inline examples/Normal_Reflectivity/Custom_XY/orsoCustomXYDSPC.m:2]: typo - "supported"

File: examples/Normal_Reflectivity/README.md [hunk 13]
```diff
@@ -1,31 +1,23 @@
-
-
-## 'Standard Target Functions' Examples
+## 'Non-polarised Target Functions' Examples
 
 This folder contains examples of how to use RAT to analyse what are referred
-to and 'Standard TF' Samples. This refers to reflectivity data that does not use
+to as 'Non-polarised TF' Samples. This refers to reflectivity data that does not use
```
Comment (DrPaulSharp) [inline examples/Normal_Reflectivity/README.md:4]: "samples" should be lower case here.

**Review by DrPaulSharp — APPROVED** [review]
Seems fine, just a few specific things to look at. I'd also ask again about the use of snake case for the directory names in the examples directory - should these not be camel case like  the rest of the code?

**Comment by StephenNneji** [comment]
> Seems fine, just a few specific things to look at. I'd also ask again about the use of snake case for the directory names in the examples directory - should these not be camel case like the rest of the code?

I left it snake case for readability but folder names are simple enough so I will change to camel case for consistency.

**Pull request merged**